bug: agent overlay anchor drift — auxiliary_client substitution fails on v2026.5.29.2

## Bug report (user-reported, 2026-06-01)

**Severity:** P0 — Fox boots without agent monkey-patches (target_model fix, auxiliary.default fallback, cron failure diagnostics). Core agent functionality degraded.

**Cause:** PR #458 bumped hermes-agent from v2026.5.16 to v2026.5.29.2. \`check-overlay-basis.sh\` validated static patches (git apply) but does NOT check runtime substitution anchors in \`fox_overlay/agent_plugins/\`. The anchor for \`auxiliary_client._get_auxiliary_task_config\` drifted — upstream refactored the function's docstring.

**Error:**
\`\`\`
AssertionError: [fox-overlay] patch agent.auxiliary_client._get_auxiliary_task_config
substitution #1: anchor expected EXACTLY ONCE, found 0x.
Upstream may have refactored this function — refresh the anchor in the
fox-overlay patch module.
Anchor: """Return the config dict for auxiliary.<task>, or {} when unavailable."""
\`\`\`

**Impact:** Gateway boots without Fox overlay patches. The warning is non-fatal (gateway runs vanilla) but the user loses:
- target_model fix
- auxiliary.default fallback
- cron failure diagnostics

**User also reported:**
- Tailscale hung installation (had to run \`tailscale login\` manually)
- \`hermes doctor --fix\` suggests \`source venv/bin/activate\` but container uses \`/bin/sh\` which doesn't have \`source\` (should be \`. venv/bin/activate\`)

**Fix:** Refresh the anchor in \`fox_overlay/agent_plugins/fox_overlay_plugin/monkey_patches/auxiliary_client.py\` against agent v2026.5.29.2. Same class of fix as PR #448 (streaming.py anchor refresh for webui).

**Process gap:** \`check-overlay-basis.sh\` must be extended to also validate runtime substitution anchors, not just static patches. This is the agent-side equivalent of the streaming anchor that drifted in #443.

--- a/packages/fox-overlay/fox_overlay/agent_plugins/fox_overlay_plugin/monkey_patches/auxiliary_client.py
+++ b/packages/fox-overlay/fox_overlay/agent_plugins/fox_overlay_plugin/monkey_patches/auxiliary_client.py
@@ -53,31 +53,30 @@ def apply() -> None:
         sentinel="_fox_patched_provider_auto_fallback",
     )
 
-    # 2) _get_auxiliary_task_config: docstring + trailing logic
+    # 2) _get_auxiliary_task_config: layer auxiliary.default underneath task config.
+    # Refreshed for v2026.5.29.2: upstream added multi-line docstring and
+    # plugin-declared defaults (register_auxiliary_task). Fox's auxiliary.default
+    # fallback is a different feature (reads from config.yaml, not plugins).
+    # We insert it between the config read and the plugin-defaults layering.
     substitute_function(
         upstream_module=_u,
         function_name="_get_auxiliary_task_config",
         substitutions=[
             (
-                '    """Return the config dict for auxiliary.<task>, or {} when unavailable."""\n',
-                '    """Return the config dict for auxiliary.<task>, or {} when unavailable.\n'
-                '\n'
-                '    Falls back to auxiliary.default when no task-specific config exists.\n'
-                '    Task-specific keys win over default keys ({**default_config, **task_config}).\n'
-                '    """\n',
-            ),
-            (
                 '    aux = config.get("auxiliary", {}) if isinstance(config, dict) else {}\n'
-                '    task_config = aux.get(task, {}) if isinstance(aux, dict) else {}\n'
-                '    return task_config if isinstance(task_config, dict) else {}\n',
+                '    task_config = aux.get(task, {}) if isinstance(aux, dict) else {}\n',
                 '    aux = config.get("auxiliary", {}) if isinstance(config, dict) else {}\n'
-                '    if not isinstance(aux, dict):\n'
-                '        return {}\n'
-                '    default_config = aux.get("default", {})\n'
-                '    default_config = default_config if isinstance(default_config, dict) else {}\n'
-                '    task_config = aux.get(task, {}) if isinstance(aux, dict) else {}\n'
-                '    task_config = task_config if isinstance(task_config, dict) else {}\n'
-                '    return {**default_config, **task_config}\n',
+                '    # Fox patch: layer auxiliary.default underneath task-specific config\n'
+                '    if isinstance(aux, dict):\n'
+                '        _default_config = aux.get("default", {})\n'
+                '        if isinstance(_default_config, dict) and _default_config:\n'
+                '            _task_raw = aux.get(task, {}) if isinstance(aux, dict) else {}\n'
+                '            _task_raw = _task_raw if isinstance(_task_raw, dict) else {}\n'
+                '            task_config = {**_default_config, **_task_raw}\n'
+                '        else:\n'
+                '            task_config = aux.get(task, {}) if isinstance(aux, dict) else {}\n'
+                '    else:\n'
+                '        task_config = {}\n',
             ),
         ],
         sentinel="_fox_patched_auxiliary_default_fallback",

